pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -221,12 +222,20 @@ protected void internalDeleteNamespace(AsyncResponse asyncResponse, boolean auth
         boolean isEmpty;
         List<String> topics;
         try {
+            int timeoutSeconds = config().getZooKeeperOperationTimeoutSeconds();
             topics = pulsar().getNamespaceService().getListOfPersistentTopics(namespaceName)
-                    .get(config().getZooKeeperOperationTimeoutSeconds(), TimeUnit.SECONDS);
-            topics.addAll(getPartitionedTopicList(TopicDomain.persistent));
-            topics.addAll(getPartitionedTopicList(TopicDomain.non_persistent));
+                    .get(timeoutSeconds, TimeUnit.SECONDS);
```

Comment:
Not async?

Reply:
This is not the focus of this refactoring, it just affects here. And will keep the original code logic.